library/src/main/java/org/apache/apex/malhar/lib/state/managed/IncrementalCheckpointManager.java

```diff
@@ -109,8 +110,8 @@ public void run()
           transferWindowFiles();
           if (latestExpiredTimeBucket.get() > -1) {
             try {
-              managedStateContext.getBucketsFileSystem().deleteTimeBucketsLessThanEqualTo(
-                  latestExpiredTimeBucket.getAndSet(-1));
+              latestPurgedTimeBucket = latestExpiredTimeBucket.getAndSet(-1);
```

Comment:
I see what you're saying. However, deleting time buckets on a separate thread can also be a problem with idempotency, right? The intention of the change here is to record the latest purged time bucket so that the transferWindowFIles call will not try to transfer the deleted time buckets.